### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by dsherma7** [n-e-sherman/mps#2]
@n-e-sherman 
This is pretty impressive bruh. Though I did have some comments below

**Specific Issues:**

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/infrastructure/factory.h#L22
Why is this `PascalCase`, but the other factories are `camelCase`?

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/infrastructure/util.h#L68
This is the prettiest code I've ever seen, lol

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/model/heisenberglouiville.h#L20
Even though you're using `auto`, variable names like `b` is bad coding practice. This should be something like
```c
for(auto bond : *lattice)
{
	if(bond.type == Lattice::physical)
	{
            ampo += 0.5,"S+",b.s1,"S-",b.s2;
            ampo += 0.5,"S-",b.s1,"S+",b.s2;
            ampo +=     "Sz",b.s1,"Sz",b.s2;
	}
	...
}
```

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/lattice/lattice.h#L58
Get rid of commented code

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/lattice/lattice.h#L40
Why do you have two different `public` sections?

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/lattice/lattice.h#L9
I think you're not supposed to use namespaces. Instead use `namespace::thing::member`

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/repository/repositorybuilder.h#L11
Do you need to specify default constructor/destructors?

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/services/spectralbroadening.h#L151

Don't use the variable name `temp`.

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/services/service.h#L9

This should probably be called something like `Service`

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/state/statebuilder.h#L27

Enums should be defined outside a class so someone else can refer to them. Probably.

https://github.com/n-e-sherman/mps/blob/21b4bf8196a990533cba07ce6b033961834cf1b3/state/groundstate.h#L22

You're indentation and spacing is garbage, 😄 

**Other Comments:**
- All your functions/methods/classes need docstrings
- Your file names should be `snake_case`, something like `heisenberglouiville.h` is hard to read.
- Your spacing is all over the place. Sometimes you have two newlines before `public`, `protected`, etc. sometimes 1 and sometimes 0. Pick one and stick to it.
- Basically remove the `scripts` directory. That's really amateur and is not meaningful outside the scope of your specific experiment.
- You're supposed to have all members of a class in the order `public`, `protected`, `private`.

**Structure:**
The structure of your code should be
```
mps/
+-- Makefile
+-- .gitignore
+-- src/
    +-- krylov/
    +-- model/
    +-- lattice/
    ...
+-- build/
+-- tests/
```
Then you could have tests in `tests/` and `build/` can be where your `Makefile` puts any `.o`s or something and anything in root should be things like `.gitignore` or `.gitattributes`, `dockerfiles`, or `.yaml` files.
